### ISSUE TITLE
[20.04] Empty Trash function doesn't work

### ISSUE THREAD

**Issue by CommunityHelper6** [pop-os/desktop-icons-ng#17]
**(1) Issue/Bug Description:**

Empty Trash function doesn't work when using Trash Icon added to desktop

**(2) Steps to reproduce (if you know):**

1.  Right click on Desktop away from other icons then click on "Desktop Icon Settings"

2.  Turn On "Show the trash icon in the desktop"

3.  If no files exist in trash, delete a test file so the icon shows that there's trash

4.  Right click on Trash Icon then click on "Empty Trash"

5.  The pop up menu disappears and nothing happens.  Trash remains.

**(3) Expected behavior:**

To have Trash files cleared

**(4) Distribution (run `cat /etc/os-release`):**
 
NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os

**(5) Gnome Shell version:**

3.36.8  (latest updates as of 18 Aug 2021)

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
<!--
````pop-shell:
  Installed: 1.1.0~1628527572~20.04~3eda9bb
  Candidate: 1.1.0~1628527572~20.04~3eda9bb
  Version table:
 *** 1.1.0~1628527572~20.04~3eda9bb 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main amd64 Packages
       1001 http://ppa.launchpad.net/system76/pop/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
```
-->



**(7) Where was Pop Shell installed from:**

Installed by default by Pop OS ISO

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**

1 x 1080p

**(9) Other Installed/Enabled Extensions:**

Dash to Panel  (issue was present beforehand)

**(10) Other Notes:**

Thanks to all devs and contributors for all their hard work!   I hope it will be an easy fix as it works using the "Files" app

**Comment by CommunityHelper6** [comment]
@jacobgkau 

Thank you for moving the submissions to the correct group and for verifying that the issues could be recreated.

Since it's working in 21.04 (Gnome 3.38), would the fix be back-ported to 20.04 LTS?  Such a basic function needs to work in any distro, particularly an LTS.

**Comment by mmstick** [comment]
Newer versions of the extension do not support GNOME 3.36.

**Comment by CommunityHelper6** [comment]
@mmstick 

Please note my comment on the extension portion:
"Dash to Panel (issue was present beforehand)"

As well as @jacobgkau  commenting:  "I can recreate both of these issues on Pop!_OS 20.04 LTS (GNOME 3.36), but not 21.04 (GNOME 3.38)."

The issues have nothing to do with "Newer versions of the extension" not supporting "GNOME 3.36."

**Comment by CommunityHelper6** [comment]
It's interesting on how an integral part of the OS is an extension.   

@jacobgkau   Any idea on how long this might take to fix?  If version GNOME 3.36 needs to be upgraded to 3.38, will that be done for the LTS any time soon?   

These are the kind of simple, but crucial, issues that shouldn't be happening in an LTS.   It makes it harder to transition basic Windows users over to Linux when emptying the Trash Bin requires steps beyond the ones they're used to, etc. 

Pop! OS has been great entry into the Linux Distro world and and hope they can keep up the momentum!

Thanks  again.

**Comment by jacobgkau** [comment]
@CommunityHelper6 The LTS would not get updated to GNOME 3.38. It's worth noting, you should be using the latest version of Pop!_OS unless you rely on software that specifically requires an LTS base.

@mmstick @ids1024 I think we should either work on fixing these issues (hopefully as simple as https://github.com/pop-os/desktop-icons-ng/pull/14 was) or roll back the extension version on Focal to what it was before https://github.com/pop-os/desktop-icons-ng/pull/11.

**Comment by CommunityHelper6** [comment]
@jacobgkau   Thank you for the prompt responses.  I would hope the LTS would get an update, however possible, as the LTS version is something I would help clients and friends alike install instead of a rolling release.   Stability is key in those situations, much like server roll-outs.

Thanks to everyone else involved in the dev work for Pop!_OS!

**Comment by CommunityHelper6** [comment]
> @jacobgkau Thank you for the prompt responses. I would hope the LTS would get an update, however possible, as the LTS version is something I would help clients and friends alike install instead of a rolling release. Stability is key in those situations, much like server roll-outs.
> 
> Thanks to everyone else involved in the dev work for Pop!_OS!

@jacobgkau   In response to your emoji:   Plain and simple - An LTS shouldn't have such issues for a frequently used function such as the Trash and should be able to be fixed without major hurdles.

**Comment by mmstick** [comment]
This will be fixed soon with a revert for Focal

**Comment by CommunityHelper6** [comment]
Much appreciated  @mmstick   @jacobgkau

**Comment by jacobgkau** [comment]
Confirmed that dragging to trash and emptying trash both work again on 623022b, which is now published on 20.04.

**Comment by CommunityHelper6** [comment]
@jacobgkau   I can also confirm the fix after the version regression.    Although it doesn't mention it after the update, a reboot is necessary but that's the least of the worries.   It works!  :smile:   Thanks!
